internal/manifest/plan.go

```diff
@@ -0,0 +1,432 @@
+package manifest
```

Comment:
Even GitHub thinks this file is too big to review - it hides the diff by default :)

But I don't have a good suggestion for splitting it...